flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/DispatcherResourceCleanupTest.java

```diff
@@ -651,6 +661,104 @@ public void testFailingJobManagerRunnerCleanup() throws Exception {
         awaitStatus(dispatcherGateway, jobId, JobStatus.RUNNING);
     }
 
+    @Test(timeout = 5000L)
+    public void testArchiveSuccessfullyWithinTimeout() throws Exception {
```

Comment:
Could we try to merge the test cases to remove redundant code? They look quite alike except for the part where we specify the final state of the job...

Reply:
Thank you for pointing this out. I've tried to cleanup the duplicated codes in these tests.